### ISSUE TITLE
[BUG]: When Selecting "All" in Dashboard Filter, "All" should mean all data in dataset matching items listed in filter, not all data in dataset

### ISSUE THREAD

**Issue by agroton-libtax** [RevealBi/Reveal.Sdk#84]
### SDK Version

1.6.0

### Client Framework

Angular

### Server Platform

ASP.NET

### Operating System

Windows

### Description

Background: 
In our business domain, a user belongs to 0 or more Offices. We have created a dashboard with a GA4 data source, its purpose is to show the user Google Analytics marketing data related to their Offices. We want to filter the data based on Page Path, and added a Dashboard filter for this, but there are over 3000 values, most of which are not relevant. We really just want to list the page paths which contain Office IDs of Offices the user belongs to. 

I set up an In Memory data source in our ASP.NET API, that uses the UserContextProvider to grab the user's Offices and generates a list of Page Paths the user should see. I wired up a new Dashboard filter to this in memory data source. I then made a copy of the first widget on the dashboard to use for testing. The original widget is connected to the original Page Path filter, and the copied widget is connected to the new filter using my in memory data source.

Problem:
When I select "All" in my new filter (which auto-selects the 4 individual items) and individually select the same 4 items in the original Page Path filter, the results shown in their respective connected widgets do not match.

Expected Results:
Selecting "All" in filter dropdown should return only data with page paths matching any of the individual items listed in the filter.

Actual Results:
Selecting "All" in filter dropdown returns entire unfiltered dataset, even items that don't match any of the individual items listed in the filter. As a user, if I have a filter dropdown and my only options are: _All, Dog, Cat_; and I select "All", I am not expecting to see "Bird" in the results.

https://github.com/RevealBi/Reveal.Sdk/assets/136626707/d8d651ff-e0e7-4ee6-a4d5-363391fe89f9

**Comment by brianlagunas** [comment]
This is not really a bug. The filter's "All" options doesn't send each filter value to the data source, it actually doesn't send any filter values, but rather queries the entire data source. So even if you customize the filter drop down, the "All" option would not use each individual filter value.

What you're really asking for is the ability to have the filter respect the same user context filtering that the data source supports. Is this correct?

**Comment by brianlagunas** [comment]
Is this the same or related to your issue #85

**Comment by agroton-libtax** [comment]
> This is not really a bug. The filter's "All" options doesn't send each filter value to the data source, it actually doesn't send any filter values, but rather queries the entire data source. So even if you customize the filter drop down, the "All" option would not use each individual filter value.
> 
> What you're really asking for is the ability to have the filter respect the same user context filtering that the data source supports. Is this correct?

I don't think so. We are not doing any user context filtering with the GA4 data source. We don't have any control over the configuration of the account and as far as I can see, there is no way for us to programmatically indicate in DataSourceProvider that we only want data where the Page Path contains an Office ID belonging to the user (correct me if I'm wrong).

I see what you're saying though, about "All" querying the entire data source. So then I guess I would want a way to configure that "All" option to either mean "entire dataset" or "all filters", or at the very least, a way to indicate I don't want to show the "All" option even if I have multiple select enabled.

**Comment by agroton-libtax** [comment]
> Is this the same or related to your issue #85

It is related, yes.
